.github/workflows/main.yml

```diff
@@ -0,0 +1,156 @@
+---
+
+name: main
+
+on:
+  workflow_dispatch:
+    inputs:
+      saucelab_test:
+        description: 'Enable run a Sauce lab test'
+        default: true
+        required: false
+        type: boolean
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+
+# Limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+  statuses: write
+
+env:
+  DOCKER_ELASTIC_SECRET: secret/observability-team/ci/docker-registry/prod
+  SAUCELABS_SECRET: secret/apm-team/ci/apm-agent-rum-saucelabs@elastic/apm-rum
+  CODECOV_SECRET: secret/apm-team/ci/apm-agent-rum-codecov
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+      - run: '.ci/scripts/lint.sh'
+      - run: 'npm run ci:bundlesize'
+        id: bundlesize
+      - uses: actions/github-script@v6
+        env:
+          BUNDLESIZE_PASS: ${{ steps.bundlesize.outputs.bundlesize_pass }}
+          BUNDLESIZE_FAIL: ${{ steps.bundlesize.outputs.bundlesize_fail }}
+        with:
+          script: |
+            const { BUNDLESIZE_PASS, BUNDLESIZE_FAIL } = process.env;
+            let state = 'success';
+            if (BUNDLESIZE_FAIL > 0) {
+              state = 'failure';
+            }
```

Comment:
NIT, can `${{ steps.bundlesize.outputs.bundlesize_pass }}` be used within the `script` context? something like


```
...
context: `${context.workflow} / Bundlesize: ${{ steps.bundlesize.outputs.bundlesize_pass }} pass / ${{ steps.bundlesize.outputs.bundlesize_fail }} fail`,
```

Reply:
It seems possible to use workflow variables within the generated script.
However, I prefer to use env vars everywhere as recommended in the [documentation](https://github.com/actions/github-script#use-env-as-input).